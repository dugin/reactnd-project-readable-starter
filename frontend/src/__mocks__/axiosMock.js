import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import {categories, posts} from "./responseMock";

const mock = new MockAdapter(axios);

const URL = 'http://localhost:3001';

mock.onGet(URL + '/categories').reply(200, categories);

mock.onGet(URL + '/posts').reply(200, posts);
