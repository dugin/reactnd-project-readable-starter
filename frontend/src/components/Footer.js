import React from "react";
import styled from "styled-components";

const StyledParagraph = styled.p`
  width: 100%;
  text-align: center;
  font-weight: 300;
  margin-bottom: 20px;
`;

const Footer = () => {
  return <StyledParagraph>Developed by Rodrigo Dugin</StyledParagraph>;
};

export default Footer;
