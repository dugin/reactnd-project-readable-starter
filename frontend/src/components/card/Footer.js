import moment from "moment/moment";
import React from "react";
import styled from "styled-components";
import colors from "../../styles/constants/colors";
import { ModeComment } from "material-ui-icons";
import { StyledPrimaryChip } from "../../styles/styles";
import { TYPE } from "../../utils/constants";

const StyledFooter = styled.section`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-left: -10px;
  margin-top: 10px;
  p {
    margin: 0 0 0 10px;
  }
`;

const StyledFooterContent = styled.div`
  display: flex;
  flex-grow: 1;
  align-items: center;
`;

const StyledFooterComments = styled.div`
  position: relative;
  height: 40px;
  width: 40px;
  margin-right: 25px;
  svg {
    height: 100%;
    width: 100%;
    color: ${colors.primary};
  }
  span {
    font-size: 12px;
    position: absolute;
    left: 0;
    bottom: 3px;
    height: 100%;
    width: 100%;
    color: white;
    display: flex;
    align-items: center;
    justify-content: center;
  }
`;

const StyledSecondaryChip = StyledPrimaryChip.extend`
  background-color: ${colors.secondary};
`;

const Footer = ({ author, timestamp, voteScore, type, amount }) => {
  const fromNow = () => {
    return moment(timestamp).fromNow();
  };

  return (
    <StyledFooter>
      <StyledFooterContent>
        <StyledSecondaryChip label={voteScore} />
        <p>
          by <b>{author}</b>
        </p>
        <p>{fromNow()}</p>
      </StyledFooterContent>

      <StyledFooterComments>
        {type !== TYPE.comment && (
          <div>
            <ModeComment />
            <span> {amount} </span>
          </div>
        )}
      </StyledFooterComments>
    </StyledFooter>
  );
};

export default Footer;
