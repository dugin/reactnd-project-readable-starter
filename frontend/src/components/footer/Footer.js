import React, {Component} from 'react';
import './Footer.css';

class Footer extends Component {

    render() {
        return (
            <div className="Footer">
                <p className="Footer__text">Developed by Rodrigo Dugin</p>
            </div>
        )
    }
}

export default Footer;