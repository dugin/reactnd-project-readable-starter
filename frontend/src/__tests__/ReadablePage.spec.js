import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Footer from "../components/footer/Footer";
import ReadablePage from "../components/readable-page/Readable.page";
import '../__mocks__/axiosMock';

describe('ReadablePage  Component', () => {

    const readablePage = shallow(<ReadablePage/>);

    it('renders properly', () => {

        expect(toJson(readablePage)).toMatchSnapshot();

    });
});