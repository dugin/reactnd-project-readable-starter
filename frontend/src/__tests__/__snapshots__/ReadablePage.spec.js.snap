// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReadablePage  Component renders properly 1`] = `
<div
  className="ReadablePage"
>
  <Categories />
  <SortBy />
  <div
    className="ReadablePage__add"
  >
    <Link
      className="btn ReadablePage__add--btn rounded-circle btn-success"
      replace={false}
      to="/create"
    >
      <IoPlusRound
        size={25}
      />
    </Link>
  </div>
</div>
`;
